odl/operator/operator.py

```diff
@@ -206,29 +395,64 @@ class Operator(with_metaclass(_OperatorMeta, object)):
     implementation of the respective other is provided.
     """
 
+    def __new__(cls, *args, **kwargs):
+        """Create a new instance."""
+        instance = super().__new__(cls)
+
+        call_has_out, call_out_optional, _ = _dispatch_call_args(cls)
+        instance._call_has_out = call_has_out
+        instance._call_out_optional = call_out_optional
+        if not call_has_out:
+            # Out-of-place _call
+            instance._call_in_place = partial(_default_call_in_place,
```

Comment:
Why do we create two new attributes (_call_in_place, _call_out_of_place) instead of simply wrapping _call with an extra optional out parameter? Wouldn't it be less complicated?